game/lib/gameplay/camera_page.dart

```diff
@@ -0,0 +1,168 @@
+            break;
+        }
+      }
+    });
+  }
+
+  toggleCamera() async {
+    int newSelected = (_selected + 1) % _cameras.length;
+    _selected = newSelected;
+
+    var controller =
+        await CameraController(_cameras[_selected], ResolutionPreset.max);
+    setState(() => _controller = controller);
+  }
+
+  void takePhoto() async {
+    try {
+      final image = await _controller.takePicture();
+      setState(() {
+        imagePath = image.path;
+        //TODO: save image path to backend
+      });
+    } catch (e) {
+      print(e);
+    }
+  }
+
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Container(
+        padding: EdgeInsets.only(top: 80.0),
+        decoration: BoxDecoration(
+          gradient: LinearGradient(
+            colors: [
+              Color.fromARGB(255, 237, 86, 86),
+              Color.fromARGB(77, 237, 86, 86)
+            ], // Adjust colors as needed
+            begin: Alignment.topCenter,
+            end: Alignment.bottomCenter,
+            stops: [0.0, 0.5], // Adjust stops as needed
+            tileMode: TileMode.clamp,
+          ),
+        ),
+        child: Column(
+          children: [
+            Row(
+              children: [
+                TextButton(
+                  onPressed: () {
+                    Navigator.pop(context);
+                  },
+                  style: TextButton.styleFrom(
+                      backgroundColor: Color.fromARGB(0, 0, 0, 0)),
+                  child: Row(
+                    children: [
+                      Padding(
+                        padding: EdgeInsets.only(right: 8.0),
+                        child: SvgPicture.asset(
+                          "assets/icons/back.svg",
+                          width: 6,
+                          height: 14,
+                          colorFilter: ColorFilter.mode(
+                              Color.fromARGB(255, 96, 67, 91), BlendMode.srcIn),
+                        ),
+                      ),
+                      Text(
+                        "Leave Photo",
+                        style: TextStyle(
+                          fontSize: 14,
+                          fontWeight: FontWeight.w400,
+                        ),
+                      ),
+                    ],
+                  ),
+                ),
+                Spacer(),
+                ElevatedButton(
+                  onPressed: () {},
+                  child: Text(
+                    "Challenge",
+                    style: TextStyle(
+                      fontSize: 12,
+                      fontWeight: FontWeight.w400,
+                    ),
+                  ),
+                )
+              ],
+            ),
+            Text(
+              "Remember this moment with a photo!",
+              style: TextStyle(
+                fontSize: 18,
+                fontWeight: FontWeight.w700,
+              ),
+            ),
+            imagePath == ""
```

Comment:
It looks like you can use a better state here instead of imagePath == "". It's confusing to see why an empty image path means we have a camera preview. If you plan to keep this, please add a comment explaining why you make this comparison.